wp-includes/class-wp-https-detection.php

```diff
@@ -25,77 +32,95 @@ class WP_HTTPS_Detection {
 	const HTTPS_SUPPORT_OPTION_NAME = 'is_https_supported';
 
 	/**
-	 * The query var key for HTTPS detection, used to verify that the request is from the same origin.
+	 * The option name for the insecure content.
 	 *
 	 * @var string
 	 */
-	const REQUEST_QUERY_VAR = 'https_detection_token';
+	const INSECURE_CONTENT_OPTION_NAME = 'insecure_content';
 
 	/**
-	 * The number passed to the query var.
+	 * The tag names for insecure content
 	 *
-	 * @var int
+	 * @see https://developer.mozilla.org/en-US/docs/Web/Security/Mixed_content
+	 * @var array
 	 */
-	public $query_var_number;
+	public $insecure_content_tags = array(
+		'img',
+		'audio',
+		'video',
+		'script',
+		'link',
+		'iframe',
+	);
 
 	/**
 	 * Inits the class.
 	 */
 	public function init() {
 		add_action( 'wp', array( $this, 'schedule_cron' ) );
```

Comment:
@kienstra Is the idea here to _actively continue checking_ after the cron time regardless if HTTPS is enabled or not? Curious....

Reply:
Good point. This does keep checking even when HTTPS is enabled. Let me work on that.

Reply:
Hi @hellofromtonya, https://github.com/xwp/pwa-wp/pull/36/commits/b7b06f5de738b7f028a47792dfe8c0d12ad81b95 prevents running the cron hook to update the HTTPS options if the `home` and `siteurl` are already HTTPS. The UI that `WP_HTTPS_UI` adds wouldn't show in that case.

Reply:
@kienstra What if we also wrap the cron hooks within a conditional check to see if 

```
if ( ! $this->is_currently_https() ) {
    // hooks here.
}
```

Then we only hook if HTTPS is not enabled.

Reply:
Thanks, good idea. Is https://github.com/xwp/pwa-wp/pull/36/commits/07ce3cf050fb4f65eeb0965b8a048a70355803d1 what you had in mind?